### PR TITLE
Improve escape deprecation message

### PR DESCRIPTION
**Pull request by ruudk**
This was introduced in https://github.com/twigphp/Twig/pull/4176

According to #4343 the message can be improved.

I made a few changes to the test to make it easier to work with. Best to review this PR **commit by commit**.

/cc @stof

### CODE REVIEW

File: src/Lexer.php [hunk 1]
```diff
@@ -418,7 +418,7 @@ private function stripcslashes(string $str, string $quoteType): string
                 $result .= $nextChar;
             } elseif ("'" === $nextChar || '"' === $nextChar) {
                 if ($nextChar !== $quoteType) {
-                    trigger_deprecation('twig/twig', '3.12', 'Character "%s" at position %d in string on line %d should not be escaped; the "\" character is ignored in Twig v3 but will not be in v4. Please remove the extra "\" character.', $nextChar, $i + 1, $this->lineno);
+                    trigger_deprecation('twig/twig', '3.12', 'Character "%s" at position %d in string on line %d should not be escaped; the "\" character is ignored in Twig v3 but will not be in v4. Please remove the extra "\" character or add another one if you want to print the escape character too.', $nextChar, $i + 1, $this->lineno);
```
Comment (stof) [inline src/Lexer.php]: I'm not sure we need this alternative, as the deprecation generally explains how to keep the same behavior.

File: tests/LexerTest.php [hunk 2]
```diff
@@ -268,6 +276,15 @@ public function testStringWithEscapedDelimiterProducingDeprecation(string $templ
 
     public static function getStringWithEscapedDelimiterProducingDeprecation()
     {
+        yield [
+            <<<'EOF'
+            {{ 'Ymd\THis' }}
```
Comment (stof) [inline tests/LexerTest.php]: this case is exactly the same one that the next case, so no need to add it.

File: src/Lexer.php (hunk 1 above)
Comment (ruudk) [reply src/Lexer.php]: Alright, removed it.

File: tests/LexerTest.php (hunk 2 above)
Comment (ruudk) [reply tests/LexerTest.php]: Removed it.

**Review by stof — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ruudk.

**Comment by smnandre** [comment]
Would it be possible to add the source path (if it exists) ?   `$this->source?->getPath()` ? 

It took me time to figure where one came from in my code (spoiler alert: a date format 😅)

**Pull request merged**